### PR TITLE
Bug 1577769 - Add support for outputting broker catalog request in yaml, json

### PR DESCRIPTION
**Pull request by dymurray**

### CODE REVIEW

File: cmd/broker.go [hunk 1]
```diff
@@ -64,12 +69,20 @@ func init() {
 	brokerCmd.PersistentFlags().StringVarP(&brokerName, "name", "n", "", "Name of Automation Broker instance")
 	rootCmd.AddCommand(brokerCmd)
 
+	brokerCatalogCmd.Flags().StringVarP(&catalogOutputFormat, "output", "o", "", "Display broker catalog output in JSON or YaML format")
```
Comment (jmrodri) [inline cmd/broker.go]: Is it really YaML? The YAML site seems to always use YAML. http://yaml.org/

File: cmd/broker.go [hunk 2]
```diff
@@ -122,7 +135,15 @@ func listBrokerCatalog(brokerRouteName string, brokerNamespace string) {
 		log.Errorf("Failed fetch catalog: %v", err)
 		return
 	}
-	printServices(services.Services)
+
+	switch catalogOutputFormat {
+	case "json":
+		printServicesAsJSON(services.Services)
+	case "yaml":
+		printServicesAsYaML(services.Services)
```
Comment (jmrodri) [inline cmd/broker.go]: Again should this be YAML instead of YaML?

File: cmd/broker.go [hunk 3]
```diff
@@ -196,7 +217,34 @@ func bootstrapBroker(brokerRouteName string, brokerNamespace string) {
...
+	encoder.SetIndent("", "    ")
+
+	err := encoder.Encode(services)
+	if err != nil {
+		log.Errorf("Failed to encode services data: [%v]", err)
+		return
+	}
+	fmt.Printf("%v", buffer.String())
+	return
+}
```
Comment (jmrodri) [inline cmd/broker.go]: I don't care for the duplication, we could remove it with an interface:
```golang
type ServiceEncoder interface {
    Encode(interface{}) error
}

func printServices(enc *ServiceEncoder) {
}
```

File: cmd/broker.go (hunk 1 above)
Comment (dymurray) [reply cmd/broker.go]: Oh you're right.

File: cmd/broker.go [hunk 4]
```diff
@@ -64,12 +73,20 @@ func init() {
...
+	brokerCatalogCmd.Flags().StringVarP(&catalogOutputFormat, "output", "o", "", "Display broker catalog output in JSON or YAML format")
 	brokerCmd.AddCommand(brokerCatalogCmd)
 
 	brokerCmd.AddCommand(brokerBootstrapCmd)
 	rootCmd.AddCommand(createHiddenCmd(brokerBootstrapCmd, "running 'apb broker bootstrap'"))
 }
 
+func pruneOutputFormat(format string) {
+	if format != "yaml" && format != "json" && format != "" {
+		log.Warnf("Did not recognize --output argument [%v], printing as table", format)
```
Comment (djwhatle) [inline cmd/broker.go]: Could we give the user a list of acceptable output types here? Let them know that they must type either "json" or "yaml".

**Pull request merged**